src/utils/BatchManager.js

```diff
@@ -2,8 +2,13 @@ function errorToSerializable(error) {
   // istanbul ignore next
   if (error === undefined) throw new TypeError('No error was passed');
 
-  // make sure it is an error object so we can serialize it properly
-  const err = error instanceof Error ? error : new Error(error);
+  // make sure it is an object that is Error-like so we can serialize it properly
+  // if it's not an actual error then we won't create an Error so that there is no stack trace
+  // because no stack trace is better than a stack trace that is generated here.
+  const err = typeof error === 'object' && typeof error.stack === 'string'
```

Comment:
`instanceof` doesn't work across realms, like `vm` - you probably want `Object.prototype.toString.call(error) === '[object Error]' && typeof error.stack === 'string'` so as to be a bit more robust.